compaction/compaction_manager.hh

```diff
@@ -334,6 +334,10 @@ private:
     // Get candidates for compaction strategy, which are all sstables but the ones being compacted.
     std::vector<sstables::shared_sstable> get_candidates(compaction::table_state& t) const;
 
+    template <std::ranges::range Range>
+    requires std::convertible_to<std::ranges::range_value_t<Range>, sstables::shared_sstable>
+    std::vector<sstables::shared_sstable> get_candidates(compaction::table_state& t, const Range& sstables) const;
```

Comment:
Explicit instantiation isn't the same as overloads. With overloads you have to duplicate the source code.

Reply:
But why not standardize on std::vector?

Reply:
I'd like to use std::unordered_set to track sstables that need cleanup to get lookup and erase in O(1).

Reply:
ok

Reply:
I was thinking about explicit overloads and a hidden template in the `.cc` file. I dislike explicitely instantiated templates because when I see a template method I instantly think I can pass anything and I will find out at link time this is not the case. Not the end of the world, you can consider this personal preference, so I won't insist on this.

Reply:
It's a private method, so I think it's an overkill to fuss about it.